templates/repo/recent_commits.tmpl

```diff
@@ -0,0 +1,9 @@
+{{if .Permission.CanRead $.UnitTypeCode}}
+	<div id="repo-recent-commits-chart"
+		data-locale-loading-title="{{ctx.Locale.Tr "repo.recent_commits.loading_title"}}"
+		data-locale-loading-title-failed="{{ctx.Locale.Tr "repo.recent_commits.loading_title_failed"}}"
+		data-locale-loading-info="{{ctx.Locale.Tr "repo.recent_commits.loading_info"}}"
+		data-locale-component-failed-to-load="{{ctx.Locale.Tr "repo.recent_commits.component_failed_to_load"}}"
```

Comment:
Simply put it after the name of the locale key, separated with a whitespace. You can also use another locale string:

```suggestion
		data-locale-component-failed-to-load="{{ctx.Locale.Tr "graphs.component_failed_to_load" (ctx.Locale.Tr "repo.recent_commits")}}"
```

Reply:
As it looks to me like we could use such messages in various places, I'd create a separate group for that.

```ini
[graphs]
component_failed_to_load="An unexpected error happened when loading %s"
```